TAO/TAO_IDL/util/utl_global.cpp

```diff
@@ -1930,63 +1930,87 @@ struct yy_buffer_state;
 extern yy_buffer_state *tao_yy_scan_string (const char *);
 extern int tao_yylex_destroy ();
 
+namespace
+{
+  class OldState
+  {
+    public:
+      OldState(IDL_GlobalData* the_idl_global, bool the_disable_output = false)
+        : idl_global_(the_idl_global),
+          old_filename_(idl_global_->filename ()),
+          old_lineno_(idl_global_->lineno ()),
+          old_idl_src_file_(idl_global_->idl_src_file ()),
+          disable_output_(the_disable_output),
+          flags_(ACE_LOG_MSG->flags ())
+      {
+        idl_global_->set_lineno (1);
+        idl_global_->set_filename(0);
+
+        // Name this pseudo-file "builtin-N"
+#define BUILTIN_NAME_BUFFER_SIZE 64
+        static char buffer[BUILTIN_NAME_BUFFER_SIZE];
+        static unsigned n = 1;
+        ACE_OS::snprintf (&buffer[0], BUILTIN_NAME_BUFFER_SIZE, "builtin-%u", n++);
```

Comment:
Looking at this again, using a macro for this was a poor decision on my part. Since your touching this code could you get rid of the macro?

```suggestion
        ACE_OS::snprintf (&buffer[0], sizeof buffer, "builtin-%u", n++);
```

Reply:
What about?

        // Name this pseudo-file "builtin-N"
        static char buffer[64];
        static unsigned n = 1;
        ACE_OS::snprintf (&buffer[0], sizeof buffer, "builtin-%u", n++);

Reply:
Yes, that's what I had in mind.